### PR TITLE
Implement Hourly Temperature Graph

### PR DESCRIPTION
**Pull request by HaliTran**
# New Features / Changes I Made
1. Create ChartJS `HourlyTempGraph` component
2. Center the `Sun` component vertically using Flex for better responsiveness
3. Misc style and layout changes

## `HourlyTempGraph` 
- I noticed the sidebar doesn't display the temperature when the user searches for a city but rather only the time and icon of the weather. Since we have to include a graph in our dashboard, I utilized GraphJS's line graph to display the temperature for each hour. The graph is currently configured to display the temperature 10 hours ahead.

## `Sun`
- The `Sun` component wasn't centered vertically and when you decrease the screen size, the space directly below the Sun would fill up with white space that didn't look so good. I simply made the parent a flex container and centered is so that it would be more responsive and cleaner to change. This doesn't affect the layout or mess anything up with Bootstrap grid because the `Sun` component has a `col` attribute attached to it but flex only affects the child inside of the `Sun` component.

## `Misc`
- Change attribute of the Sun svg file so that the animation would rotate slightly quicker.
- Changed helper function that retrieved the hours to capitalize AM and PM.

The style and look of the graph is still tentative. We can figure this out when we decide a theme for the dashboard. I just figured blue looks okay for now!

# What Feedback I'm Looking For
Following the project guidelines, I'd appreciate any feedback on:
 - **Accessibility**
 - **Responsiveness**
 - **Airbnb style guide rules**
 - **Unnecessary code**
 - **Proper and consistent formatting**
 
***

### CODE REVIEW

File: src/components/main-dashboard/weather-details/HourlyTempGraph.jsx [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+        title: {
+          display: true,
+          text: 'Time',
+          fontSize: 12,
+        },
+      },
+      y: {
+        title: {
+          display: true,
+          text: 'Temperature (F°)',
```
Comment (awong-pdx) [inline src/components/main-dashboard/weather-details/HourlyTempGraph.jsx]: I think this should read `(°F)`

File: src/components/main-dashboard/weather-details/HourlyTempGraph.jsx [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+        backgroundColor: 'rgba(75, 192, 192, 0.5)',
+        borderWidth: 3,
+        tension: 0.4,
+      },
+    ],
+  };
+
+  return (
+    <div className="col-8 border border-primary rounded py-1">
+      <Line className="hourlyGraph" options={options} data={data} />
```
Comment (awong-pdx) [inline src/components/main-dashboard/weather-details/HourlyTempGraph.jsx]: To maintain CSS class name consistency, I think `hourlyGraph` could be changed to `hourly-graph`

File: src/utilities/helperFunctions.js [hunk 3]
```diff
@@ -48,11 +48,11 @@ export const getHourString = function getFormattedHourStringFromDate(
...
   if (cityTime < 12) {
-    if (cityTime === 0) hourString = '12 am';
-    else hourString = `${cityTime} am`;
+    if (cityTime === 0) hourString = '12 AM';
+    else hourString = `${cityTime} AM`;
   } else if (cityTime > 11) {
-    if (cityTime === 12) hourString = '12 pm';
-    else hourString = `${cityTime - 12} pm`;
+    if (cityTime === 12) hourString = '12 PM';
+    else hourString = `${cityTime - 12} PM`;
```
Comment (awong-pdx) [inline src/utilities/helperFunctions.js:55]: This is great, thank you for changing this!

File: src/components/main-dashboard/weather-details/HourlyTempGraph.jsx [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+        },
+      },
+    },
+  };
+
+  const data = {
+    labels: hourlyTimeXValues,
+    datasets: [
+      {
+        label: ' Temperature (F°)',
```
Comment (awong-pdx) [inline src/components/main-dashboard/weather-details/HourlyTempGraph.jsx]: Same as above, line 60

**Review by awong-pdx — COMMENTED** [review]
The line graph looks so good and I think it's an awesome addition! I like that it automatically scales and it seems to be pretty responsive at different scales.  

The logic is great and very readable; I just had a few comments regarding naming.

File: src/components/main-dashboard/weather-details/HourlyTempGraph.jsx (hunk 1 above)
Comment (HaliTran) [reply src/components/main-dashboard/weather-details/HourlyTempGraph.jsx]: Haha you're right, its been a long day

File: src/components/main-dashboard/weather-details/HourlyTempGraph.jsx (hunk 2 above)
Comment (HaliTran) [reply src/components/main-dashboard/weather-details/HourlyTempGraph.jsx]: Of course, thanks for catching that!

**Review by awong-pdx — APPROVED** [review]
Everything looks good, thank you!

**Pull request merged**